packages/trader/src/Modules/CFD/Containers/cfd-reset-password-modal.tsx

```diff
@@ -26,11 +27,13 @@ const ResetPasswordIntent = ({ current_list, children, is_eu, ...props }) => {
     if (has_intent && current_list) {
         [server, group, type] = reset_password_intent.split('.');
         login = current_list[`mt5.${group}.${type}@${server}`].login;
-        title = getMtCompanies(is_eu)[group][type].title;
+        title = getMtCompanies(is_eu)[group as keyof TMtCompanies][type as keyof TMtCompanies['demo' | 'real']].title;
     } else if (current_list) {
-        [server, group, type] = Object.keys(current_list).pop().split('.');
+        [server, group, type] = Object.keys(current_list)?.pop()!.split('.');
```

Comment:
Typescript exclamation mark will ignore the undefined type. It would be a better practice to handle undefined use case as well. Can we avoid using `!` here? 😊

Reply:
Since we are already checking in our condition that if (current_list) then do this line of code. So we are sure that our current list is not undefined. We need to use exclamation mark to keep TypeScript happy it will yell at us if we remove the `!` .

Reply:
@hirad-rewok unfortunately, `else if (current_list) ` doesn't protect us from `current_list` being an empty array since an empty array is a true value. That's why in case `current_list` is an empty array, the .pop() method will return `undefined`. Question is what we want to do if current_list is empty. If we want to not go further, then it's better to replace `else if (current_list) { ` `with else if (current_list.length > 0) {`      😊
And my suggestion to avoid TS yelling is:
```
} else if (current_list) {
        [server, group, type] = (Object.keys(current_list).pop() as string).split('.');
```
*I believe that `current_list` is not an object but an array with string indexes instead of numbered indexes according to the current_list() getter in cfd-store.js.

Reply:
Javascript arrays cannot have "string indexes". A Javascript Array is exclusively numerically indexed. When you set a "string index", you're setting a property of the object.

Reply:
I can use typeguard or type assertion. Since type assertion works, seems like we'd better stick to it.

Reply:
LGTM 👍🏻  BTW, For arrays, sure thing, officially a Javascript Array is exclusively numerically indexed. But you'll be surprised when you log `current_list` in the console and call array methods on it 😉
![Screenshot 2022-02-22 at 20 23 00](https://user-images.githubusercontent.com/87321516/155185196-77e9b0fe-fba2-4069-a320-1056ca5eb6d2.png)